trillian/migrillian/core/multi.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"context"
+	"sync"
+
+	"github.com/golang/glog"
+)
+
+// RunMigration migrates data from a number of CT logs to Trillian. Each log's
+// migration is coordinated by the corresponding Controller. This function
+// terminates when all Controllers are done (possibly with an erorr, or as a
+// result of canceling the passed in context).
+//
+// TODO(pavelkalinnikov):
+// - Expose status of each goroutine to metrics.
+// - Handle Controller failures, e.g. restart them (depends on error type).
+// - Introduce a MultiController type.
+// - Start Controllers with random delays to prevent one instance (e.g. the one
+//   that has started first) capturing mastership over all logs at once.
```

Comment:
I agree. This is important for mirroring but for migration less so as we don't do them all at once.

Reply:
Exactly.